google/cloud/storage/internal/grpc/configure_client_context.cc

```diff
@@ -54,13 +54,18 @@ void ApplyRoutingHeaders(
 
 void ApplyRoutingHeaders(grpc::ClientContext& context,
                          storage::internal::UploadChunkRequest const& request) {
+  ApplyResumableUploadRoutingHeader(context, request.upload_session_url());
+}
+
+void ApplyResumableUploadRoutingHeader(grpc::ClientContext& context,
+                                       std::string const& upload_id) {
   static auto* slash_format =
       new std::regex{"(projects/[^/]+/buckets/[^/]+)/.*", std::regex::optimize};
   static auto* colon_format =
       new std::regex{"(projects/[^/]+/buckets/[^:]+):.*", std::regex::optimize};
```

Comment:
Aside: didn't GCS decide not to use colons?

Reply:
#13026 I am trying to not think about this code until later.

Reply:
Ack.